packages/editor/src/store/selectors.js

```diff
@@ -99,6 +99,20 @@ export function isEditedPostNew( state ) {
 	return getCurrentPost( state ).status === 'auto-draft';
 }
 
+/**
+ * Returns true if content includes unsaved changes, or false otherwise.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {boolean} Whether content includes unsaved changes.
+ */
+export function hasChangedContent( state ) {
+	return (
+		state.editor.present.blocks.isDirty ||
+		'content' in state.editor.present.edits
```

Comment:
I don't really get what this second condition is checking for.

Reply:
I'll add a comment to clarify, but it's a combination of two things which require this consideration:

- While content in Visual mode is represented by the blocks state, in Text mode it is tracked by `edits.content`.
- `edits` is intended to contain only the different property values, so it is expected that the _mere presence_ of any property is an indicator that the value is different than what is known to be saved.